pkg/apis/core/validation/validation.go

```diff
@@ -2314,10 +2307,6 @@ func ValidateVolumeDevices(devices []core.VolumeDevice, volmounts map[string]str
 	devicepath := sets.NewString()
 	devicename := sets.NewString()
 
-	if devices != nil && !utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) {
```

Comment:
Do we need to add similar logic to drop volumeDevices too?

Reply:
yes, revert this change until that is done in a separate PR (see umbrella issue tracking all the fields that need sweeping - https://github.com/kubernetes/kubernetes/issues/72169)

Reply:
reverted